app/services/publish_service.rb

```diff
@@ -29,6 +31,20 @@ def publish_assets(live_edition)
     PublishAssetService.call(edition, live_edition)
   end
 
+  def associate_with_government
+    return if edition.government
+
+    date = edition.backdated_to || edition.document.first_published_at || Time.current
+    government = Government.for_date(date)
+    edition.assign_attributes(government_id: government&.content_id)
+  end
+
+  def sync_current_edition
```

Comment:
This method seems tightly couple with `associate_with_government`. Could they be merged?

Reply:
I'll see if the sync fits into the associate method